arangod/V8Server/v8-vocbase.cpp

```diff
@@ -1396,18 +1396,17 @@ static void MapGetVocBase(v8::Local<v8::String> const name,
 
   std::shared_ptr<arangodb::LogicalCollection> collection;
 
-  try {
-    if (ServerState::instance()->isCoordinator()) {
-      auto* ci = arangodb::ClusterInfo::instance();
+  if (ServerState::instance()->isCoordinator()) {
+    auto* ci = arangodb::ClusterInfo::instance();
 
-      collection = ci
-        ? ci->getCollection(vocbase.name(), std::string(key)) : nullptr;
+    collection = ci
+        ? ci->getCollectionNT(vocbase.name(), std::string(key)) : nullptr;
+    if (collection != nullptr) {
+      // do not propagate exception from here
+      TRI_V8_RETURN(v8::Handle<v8::Value>());
```

Comment:
additionally, we don't need this `if` and the `return` here, as the `nullptr`-case will be handled in lines 1412+.